pkg/duck/subscriber_test.go

```diff
@@ -177,9 +173,13 @@ func TestSubscriberSpec(t *testing.T) {
 					Name:      "subname",
 					Namespace: testNS,
 				},
-				Spec: v1alpha1.SubscriptionSpec{
-					Subscriber: tc.Sub,
-				},
+			}
+			if tc.Sub != nil && tc.Sub.Ref != nil {
```

Comment:
Do you need the nil check on the ref? I would expect that if I set the test case's sub, then it will be used.